### ISSUE TITLE
Future plans

### ISSUE THREAD

**Issue by notrs** [Azure-Samples/SpeechToText-WebSockets-Javascript#14]
Is there a roadmap for this project? Also, will this be adapted to allow import statements to work? 

ex:
import {SDK} from 'microsoft-speech-browser-sdk';

**Comment by RaviPidaparthi** [comment]
Thanks for asking. We will add a roadmap to the wiki soon. 

I think your question is more related to ES6?

If you are using typescript you can do imports today because typescript abstracts away the module resolution logic and lets you decide this at compile/transpile time. 
If you check our gulp file you will see that it compiles to ES5 and uses AMD module resolution.
pipe(ts({
            target: "ES5",
            declaration: true,
            noImplicitAny: true,
            removeComments: true,
            module: "AMD",
            out: 'speech.browser.sdk.js'
        }))

If you are using javascript you can't do import directly as the library is on ES5. Instead you would have to use require.js to do the module resolution.

If you really want to have ES6 modules and not depend on any external module libraries you can build the sdk yourself by targeting ES6. Just change the build task to target ES6 instead.

**Comment by notrs** [comment]
I am in a typescript project(Angular) and when trying to import the module, there is an error stating that the file is not a module

http://i.imgur.com/4peBgFr.jpg

**Comment by RaviPidaparthi** [comment]
I would have expected that to work. Let me write up a working sample and I will share that with you.

**Comment by jawetzel** [comment]
Any sort of update on this issue, or when it might get an update?

**Comment by RaviPidaparthi** [comment]
We are looking into this and will provide an update next week,

**Comment by notrs** [comment]
Thanks for looking into this Rav, is there any kind of issue that came up in trying to resolve this?

**Comment by seanfcarroll** [comment]
Is this repo the only official Microsoft SDK for Javascript? Is there a CDN anywhere hosting the SDK?

**Comment by notrs** [comment]
Any chance of getting an update on this issue? Thanks!

**Comment by priyaravi20** [comment]
Yes currently this is the only source. Thanks for feedback on CDN. We don't have plans on hosting in CDN currently.

**Comment by jawetzel** [comment]
Thanks, but i meant the original issue of this library supporting import statements that was being supported by RaviPidaparthi

**Comment by priyaravi20** [comment]
Sorry we do not have support currently. We plan to have angularjs support in a future release.

**Comment by jawetzel** [comment]
Its not just angularJs support, the ability to import a project using var package = require('package') would allow usage for any javascript frameworks including react, angularjs, and several more. My question ended up being when require will be supported as depicted in the image on post #3

**Comment by salvoravida** [comment]
look @ https://github.com/Azure-Samples/SpeechToText-WebSockets-Javascript/pull/52
